Firestore/Swift/Source/Codable/DocumentID.swift

```diff
@@ -105,12 +106,30 @@ public struct DocumentID<Value: DocumentIDWrappable & Codable>:
   private var value: Value? = nil
 
   public init(wrappedValue value: Value?) {
+    if let value = value {
+      logWarning()
+    }
     self.value = value
   }
 
-  public internal(set) var wrappedValue: Value? {
+  public var wrappedValue: Value? {
     get { value }
-    set { value = newValue }
+    set {
+      logWarning()
+      value = newValue
+    }
+  }
+
+  private func logWarning() {
```

Comment:
See @peterfriese's latest comments in API review doc. Rather than deprecate the setter, I think it may be best to log a runtime warning in the both the init and setter until we identify a longer term fix.

We can always deprecate the setter in a 10.x as we prep for Firebase 11.

Reply:
cc: @andrewheard  @paulb777